engine/core/src/main/java/org/datacleaner/reference/SimpleDictionary.java

```diff
@@ -134,10 +137,14 @@ public boolean equals(Object obj) {
 
     @Override
     public DictionaryConnection openConnection(DataCleanerConfiguration configuration) {
+        final SortedSet<String> connectionValueSet = new TreeSet<>(Comparator.comparingInt(String::length).reversed().thenComparing(String::compareTo));
```

Comment:
Shouldn't this somehow be part of consuming code - or at least somehow be revealed in the interface, because right now it takes form of a "hidden implementation detail".